### PR TITLE
Revised Release Notes

### PR DESCRIPTION
**Pull request by awholegunch**
Re-organzied and added sub-sections to "New Features".

[Staged Release Notes](http://docs-staging.cask.co/staging/cdap/3.1.0-feature-r3.1_Release_Notes_Revised/en/reference-manual/release-notes.html#new-features)

### CODE REVIEW

**Comment by abaranau** [comment]
Metrics section looks good

**Comment by sagarkapare** [comment]
Workflow section LGTM :+1:

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 1]
```diff
@@ -29,59 +29,52 @@ Cask Data Application Platform Release Notes
...
 
-- `CDAP-1982 <https://issues.cask.co/browse/CDAP-1982>`__ -
-  Added an example for Workflow fork and condition nodes.
+- `CDAP-2318 <https://issues.cask.co/browse/CDAP-2318>`__ -
+  Made CDAP work on the HDP 2.2 distribution.
 
-- `CDAP-1993 <https://issues.cask.co/browse/CDAP-1993>`__ -
-  Added ability to manipulate the SparkConf.
+- `CDAP-2786 <https://issues.cask.co/browse/CDAP-2786>`__ -
+  Added support to CDAP 3.1.0 for the MapR distro.
```
Comment (gsps1) [inline cdap-docs/reference-manual/source/release-notes.rst]: 4.1

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 2]
```diff
@@ -29,59 +29,52 @@ Cask Data Application Platform Release Notes
...
-
-- `CDAP-344 <https://issues.cask.co/browse/CDAP-344>`__ -
-  Users can now run Spark in Distributed mode.
-
-- `CDAP-1483 <https://issues.cask.co/browse/CDAP-1483>`__ -
-  QueueAdmin now uses Id.Namespace instead of simply String.
-
-- `CDAP-1584 <https://issues.cask.co/browse/CDAP-1584>`__ -
-  CDAP CLI now shows the username in the CLI prompt.
+**MapR Support, HDP 2.2 Support, CDH 5.4 Support**
```
Comment (gsps1) [inline cdap-docs/reference-manual/source/release-notes.rst]: 4.1

**Comment by henry-cask** [comment]
Distro section LGTM

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 3]
```diff
@@ -97,6 +90,18 @@ New Features
...
+  Added support for the WorkflowToken in the Spark programs.
+
+- `CDAP-2936 <https://issues.cask.co/browse/CDAP-2936>`__ -
+  Spark program can now specify resources usage for driver and executor process in distributed mode.
+
+
+**Workflows**
+
+- `CDAP-1983 <https://issues.cask.co/browse/CDAP-1983>`__ -
+  Added example application for processing and analyzing Wikipedia data using Workflows.
```
Comment (gsps1) [inline cdap-docs/reference-manual/source/release-notes.rst]: @sagarkapare shouldn't this description be about example for Workflow token usage.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 4]
```diff
@@ -29,59 +29,52 @@ Cask Data Application Platform Release Notes
...
+**Spark**
+
+- `CDAP-344 <https://issues.cask.co/browse/CDAP-344>`__ -
+  Users can now run Spark in Distributed mode.
+
+- `CDAP-1993 <https://issues.cask.co/browse/CDAP-1993>`__ -
+  Added ability to manipulate the SparkConf.
 
 - `CDAP-2700 <https://issues.cask.co/browse/CDAP-2700>`__ -
   Added the ability to Spark programs of discovering CDAP services.
```
Comment (rohitsinha54) [inline cdap-docs/reference-manual/source/release-notes.rst]: Added the ability to Spark programs of discovering CDAP services in distributed mode.

**Comment by rohitsinha54** [comment]
1 minor addition beside that LGTM.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 3 above)
Comment (sagarkapare) [inline cdap-docs/reference-manual/source/release-notes.rst]: It uses token, yes. But it also features more things as CONDITION and FORK nodes. So I think its better not to put the `WorkflowToken` only in the title.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 1 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Added.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 2 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Added.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 4 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: fixed.

**Comment by gsps1** [comment]
LGTM

**Pull request merged**